opm/simulators/flow/Main.cpp

```diff
@@ -0,0 +1,244 @@
+                    const bool init_from_restart_file,
+                    const bool allRanksDbgPrtLog,
+                    const bool strictParsing,
+                    const int mpiRank,
+                    const int output_param)
+{
+    auto omode = setupLogging(mpiRank,
+                              deckFilename,
+                              outputDir,
+                              outputMode,
+                              outputCout_, "STDOUT_LOGGER", allRanksDbgPrtLog);
+
+    if (outputCout_) {
+        OpmLog::info("Reading deck file '" + deckFilename + "'");
+    }
+
+    std::optional<int> outputInterval;
+    if (output_param >= 0)
+        outputInterval = output_param;
+
+    Opm::readDeck(EclGenericVanguard::comm(),
+                  deckFilename,
+                  eclipseState_,
+                  schedule_,
+                  udqState_,
+                  actionState_,
+                  wtestState_,
+                  summaryConfig_,
+                  std::make_shared<Python>(),
+                  strictParsing,
+                  init_from_restart_file,
+                  outputCout_,
+                  outputInterval);
+
+    verifyValidCellGeometry(EclGenericVanguard::comm(), *this->eclipseState_);
+
+    outputFiles_ = (omode != FileOutputMode::OUTPUT_NONE);
+}
+
+void Main::setupVanguard()
+{
+    EclGenericVanguard::setParams(this->setupTime_,
+                                  this->eclipseState_,
+                                  this->schedule_,
+                                  std::move(this->udqState_),
+                                  std::move(this->actionState_),
+                                  std::move(this->wtestState_),
+                                  this->summaryConfig_);
+}
+
+#if HAVE_DAMARIS
+void Main::setupDamaris(const std::string& outputDir)
+{
+    // By default EnableDamarisOutputCollective is true so all simulation results will
+    // be written into one single file for each iteration using Parallel HDF5.
+    // It set to false, FilePerCore mode is used in Damaris, then simulation results in each
+    // node are aggregated by dedicated Damaris cores and stored to separate files per Damaris core.
+    // Irrespective of mode, output is written asynchronously at the end of each timestep.
+    const bool enableDamarisOutputCollective = EWOMS_GET_PARAM(PreTypeTag, bool, EnableDamarisOutputCollective);
+    // Using the ModifyModel class to set the XML file for Damaris.
+    DamarisOutput::initializeDamaris(EclGenericVanguard::comm(), EclGenericVanguard::comm().rank(), outputDir, enableDamarisOutputCollective);
+    int is_client;
+    MPI_Comm new_comm;
+    int err = damaris_start(&is_client);
+    isSimulationRank_ = (is_client > 0);
+    if (isSimulationRank_ && err == DAMARIS_OK) {
+        damaris_client_comm_get(&new_comm);
+        EclGenericVanguard::setCommunication(std::make_unique<Parallel::Communication>(new_comm));
+    } else {
+        return false;
+    }
```

Comment:
Sorry, I didn't notice this during review.  Member function `Main::setupDamaris()` is declared as not returning anything, but the `else` branch here returns `false`.  That seems strange for two reasons,

1. The non-simulation rank, i.e., `is_client <= 0`, will return `false`
2. We'll return `false` also if `damaris_start()` returns something other than `DAMARIS_OK`.

What's the appropriate semantics here?  Granted most of the questionable logic was already there, but we need to decide what's the appropriate action for the `else` branch.

Reply:
ai. i should have picked this up but i had forgotten to enable damaris in my build configuration :/